torchtune/models/flamingo/_encoder.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional
+
+import torch
+
+from torch import nn, Tensor
+from torchtune.modules.model_fusion import register_fusion_module
+from torchtune.modules.transformer import _get_clones
+
+
+class FlamingoProjectionHead(nn.Module):
+    """Projection transformer to adapt the output of a
+    pretrained encoder to a pretrained decoder model.
+
+    Args:
+        layer (nn.Module): Transformer Decoder layer
+        num_layers (int): Number of Transformer Decoder layers
+        output (nn.Module): Output linear layer. Input dim is
+            (num_hidden + 1) * encoder_dim and output is decoder_dim.
+        num_hidden_inputs (int): Number of expected hidden state inputs
+    """
+
+    def __init__(
+        self,
+        layer: nn.Module,
+        num_layers: int,
+        output: nn.Module,
```

Comment:
should we just hardcode this to nn.Linear? I'm wondering if it makes more sense for a user to configure encoder_dim -> decoder_dim rather than the output module directly

Reply:
I can update this, I was just following our typical pattern but this would never need to be customized by a user anyway.